crates/types/src/lib.rs

```diff
@@ -25,10 +25,10 @@ pub enum WasmType {
     F64,
     /// V128 type
     V128,
-    /// FuncRef type
-    FuncRef,
-    /// ExternRef type
-    ExternRef,
+    /// Reference type
+    Ref(WasmRefType),
+    /// Bottom type
+    Bot,
```

Comment:
I think definitely for the case of the the `wasmtime` crate's public API we'll want to hide `Bot`, even if it ends up getting exposed in `wasmparser`.